### PR TITLE
Random cleanups and fixes

### PR DESCRIPTION
**Pull request by danielgustafsson**
This branch started its life intended to collect fixes to some Coverity defects but turned mostly into a set of random small cleanups encountered while perusing through the codebase.

### CODE REVIEW

**Comment by asimrp** [comment]
LGTM, although I'm not an authority on cdb_dump*.

**Comment by danielgustafsson** [comment]
Thanks, perhaps @Chibin and/or @chrishajas can have a look at the CDB pg_dump related changes?

File: src/bin/pg_dump/cdb/cdb_dump.c [hunk 1]
```diff
@@ -422,24 +382,17 @@ addChildrenToPassThrough(InputOptions *pInputOpts, char opt, SimpleOidList list)
  *
  * returns 'false' if error was reported. 'true' otherwise.
  */
```
Comment (Chibin) [inline src/bin/pg_dump/cdb/cdb_dump.c]: Description needs to get updated.

**Comment by Chibin** [comment]
@danielgustafsson We took a look into the changes and looks good at first glance.
We're currently testing out the changes.

Thanks.

File: src/bin/pg_dump/cdb/cdb_dump.c (hunk 1 above)
Comment (danielgustafsson) [inline src/bin/pg_dump/cdb/cdb_dump.c]: Can't believe I missed that.. thanks, will update

**Comment by chrishajas** [comment]
They pass all backup and restore tests. Looks good from our side.

**Comment by danielgustafsson** [comment]
Thanks for the review, have pushed a rebase on top of master and will push to master when Concourse is happy.

**Comment by danielgustafsson** [comment]
Pushed to master, thanks for review!

**Pull request closed without merging**